src/SubscribeBlockTracker.test.ts

```diff
@@ -0,0 +1,3221 @@
+                  {
+                    methodName: 'eth_subscribe',
+                    response: {
+                      result: '0x64',
+                    },
+                  },
+                ],
+              },
+            },
+            async ({ provider, blockTracker }) => {
+              const promiseForCaughtError = new Promise<any>((resolve) => {
+                blockTracker[methodToAddListener]('error', resolve);
+              });
+              blockTracker.on('_started', () => {
+                provider.emit('data', null, {
+                  method: 'eth_subscription',
+                  params: {
+                    subscription: '0x64',
+                    result: {
+                      number: '0x0',
+                    },
+                  },
+                });
+              });
+
+              const promiseForLatestBlock =
+                blockTracker[methodToGetLatestBlock]();
+
+              await expect(promiseForCaughtError).toNeverResolve();
+              const latestBlock = await promiseForLatestBlock;
+              expect(latestBlock).toBeUndefined();
+            },
+          );
+        });
+
+        it(`should emit the "error" event (added via \`${methodToAddListener}\`) and should never resolve if, while making the request for the latest block number, the provider throws an Error`, async () => {
+          recordCallsToSetTimeout({ numAutomaticCalls: 1 });
+          const thrownError = new Error('boom');
+
+          await withSubscribeBlockTracker(
+            {
+              provider: {
+                stubs: [
+                  {
+                    methodName: 'eth_blockNumber',
+                    implementation: () => {
+                      throw thrownError;
+                    },
+                  },
+                ],
+              },
+            },
+            async ({ blockTracker }) => {
+              const promiseForCaughtError = new Promise<any>((resolve) => {
+                blockTracker[methodToAddListener]('error', resolve);
+              });
+
+              const promiseForLatestBlock =
+                blockTracker[methodToGetLatestBlock]();
+
+              const caughtError = await promiseForCaughtError;
+              expect(caughtError).toBe(thrownError);
+              await expect(promiseForLatestBlock).toNeverResolve();
+            },
+          );
+        });
+
+        it(`should emit the "error" event (added via \`${methodToAddListener}\`) and should never resolve if, while making the request for the latest block number, the provider throws a string`, async () => {
+          recordCallsToSetTimeout({ numAutomaticCalls: 1 });
+          const thrownError = 'boom';
+
+          await withSubscribeBlockTracker(
+            {
+              provider: {
+                stubs: [
+                  {
+                    methodName: 'eth_blockNumber',
+                    implementation: () => {
+                      throw thrownError;
```

Comment:
Presumably this was supposed to throw a string

```suggestion
                      throw 'boom';
```

Reply:
Oh I see, the variable name is misleading here. It is a string

Reply:
Yeah, not sure if there is a better name for this. `thrownErrorMessage`? But that's misleading too. `thrownReason`?

Reply:
Maybe `thrownString`?

Reply:
Right :) Fixed in [57d0b84](https://github.com/MetaMask/eth-block-tracker/pull/103/commits/57d0b84907cf11be84b9c5620f0f7be6e8ce9a8a).